src/plonk.rs

```diff
@@ -152,16 +156,25 @@ impl<C: Curve> Circuit<C> {
 
         // Generate a random beta and gamma from the transcript.
         challenger.observe_affine_points(&c_wires);
-        let (beta, gamma) = challenger.get_2_challenges();
+        let (beta_bf, gamma_bf) = challenger.get_2_challenges();
+        let beta_sf = beta_bf.try_convert::<C::ScalarField>()?;
+        let gamma_sf = gamma_bf.try_convert::<C::ScalarField>()?;
+
+        let subgroup = C::ScalarField::cyclic_subgroup_known_order(self.subgroup_generator, self.degree());
```

Comment:
Note that this list of subgroup elements starts with `1`, so the code below will start with `Z(1) = 1`. The Plonk paper starts with `Z(g) = 1`, although I think other parts of the paper assume `Z(1) = 1`. (It's hard to tell since `L_1(x)` could refer to either the function that selects `1` or `g^1`; both come up in the paper.)

Of course either works, we just need to make sure we're consistent, and keep in mind that the paper won't always match.